test/unit/gen/certificate.go

```diff
@@ -261,3 +264,15 @@ func SetCertificateRevisionHistoryLimit(limit int32) CertificateModifier {
 		crt.Spec.RevisionHistoryLimit = &limit
 	}
 }
+
+func SetCertificateAdditionalKeyFormats(additionalKeyFormats []v1.AdditionalKeyFormat) CertificateModifier {
+	return func(crt *v1.Certificate) {
+		crt.Spec.PrivateKey.AdditionalKeyFormats = additionalKeyFormats
```

Comment:
mmh.. not sure.. the `SetCertificateKeySize` and `SetCertificateKeyAlgorithm` functions are not doing this..

Reply:
That's a good catch too @seuf - we should be doing this everywhere to avoid a nil pointer dereference :) (although we should hopefully only ever see these during test cases, so errors will be picked up long before a user gets this code!)

Reply:
The additionalOutputFomat is now at the root of the spec. So no need to check if privateKey is not nil here. (It should be test in other functions, but that's another issue :yum: )